src/shared_modules/utils/rocksDBWrapper.hpp

```diff
@@ -492,20 +492,91 @@ namespace Utils
          */
         void deleteAll() override
         {
-            // Delete from all family columns
-            for (const auto& columnHandle : m_columnsHandles)
+            auto it = m_columnsHandles.begin();
+            std::vector<std::string> columnsNames;
+
+            while (it != m_columnsHandles.end())
             {
-                rocksdb::WriteBatch batch;
-                std::unique_ptr<rocksdb::Iterator> it(m_db->NewIterator(rocksdb::ReadOptions(), columnHandle));
-                for (it->SeekToFirst(); it->Valid(); it->Next())
+                if ((*it)->GetName() != rocksdb::kDefaultColumnFamilyName)
                 {
-                    batch.Delete(it->key());
+                    auto status = m_db->DropColumnFamily(*it);
+                    if (!status.ok())
+                    {
+                        throw std::runtime_error("Error deleting data: " + status.ToString());
+                    }
+                    columnsNames.push_back((*it)->GetName());
+
+                    status = m_db->DestroyColumnFamilyHandle(*it);
+                    if (!status.ok())
+                    {
+                        std::cerr << "Failed to free RocksDB column family: " + std::string {status.getState()}
+                                  << std::endl;
+                    }
+
+                    it = m_columnsHandles.erase(it);
```

Comment:
Re-generate the column here to better atomize the method, if we fail to remove the following column we will still be in a working state
```suggestion
                    it = m_columnsHandles.erase(it);
                    createColumn(columnName);                    
```